api.go

```diff
@@ -177,6 +177,47 @@ func (r *lockedRandSource) Seed(seed int64) {
 	r.lk.Unlock()
 }
 
+// Redirect requests by re signing the request.
+func (c *Client) redirectHeaders(req *http.Request, via []*http.Request) error {
```

Comment:
Should you get rid of the redirect handling on line 488 of this file?  `mc` already checks for go `1.9.1`, so I think it's likely safe to remove.

Reply:
Yeah but minio-go is used by apps which still run on go1.7.x etc.

Reply:
Anyways removed it, most apps would be using go1.8 or 1.9

Reply:
Cool!  Thanks!